sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/subscriptions/SubscriptionsDbConnector.groovy

```diff
@@ -42,14 +43,14 @@ class SubscriptionsDbConnector implements SubscriptionDataSource, SubscribedProj
      * @since 1.1.0
      */
     @Override
-    void subscribeToProject(Subscriber subscriber, String projectCode) {
+    void subscribeToProject(Subscriber subscriber, String projectCode) throws DataSourceException{
           try {
             Connection connection = connectionProvider.connect()
             connection.setAutoCommit(false)
       
             connection.withCloseable { it ->
               try {
-                int subscriberId = getSubscriberId(it, subscriber)
+                int subscriberId = getSubscriberId(subscriber).get()
```

Comment:
That will give you an NPE, if the method returns an `Optional.empty`.

Better write it as 
```suggestion
                int subscriberId = getSubscriberId(subscriber).getOrElseThrow({new DataSourceException("some info"})
```

Reply:
you can read up more on Oracles API docs:
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
The point of Optionals is to promote the client to use save value access idioms to avoid NPE and do exception handling

Reply:
I had that first, but then I thought I just create an exception for an exception, which I found strange.